src/Components/Nav/Nav.js

```diff
@@ -1,13 +1,24 @@
+import { useLogin } from "../../contexts/login-context"
+import { Link } from "react-router-dom"
 import "./Nav.css"
+
 export const Nav = () => {
+    const { isUserLoggedIn, userInfo, logoutHandler } = useLogin();
```

Comment:
Since details.firstName is involved so you can destructure it here itself within this line
const {isUserLoggedIn,userInfo:{details:{firstName}},logoutHandler}=useLogin()